go/test/endtoend/onlineddl/scheduler/onlineddl_scheduler_test.go

```diff
@@ -991,7 +1053,679 @@ func TestSingleton(t *testing.T) {
+		uuid := testRevertMigration(t, createRevertParams(uuids[len(uuids)-1]))
+		uuids = append(uuids, uuid)
+		onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusComplete)
+		checkMigratedTable(t, tableName, "create1")
+		checkTable(t, tableName, true)
+		testSelectTableMetrics(t)
+	})
+	t.Run("declarative DROP TABLE", func(t *testing.T) {
+		// Table exists
+		uuid := testOnlineDDL(t, dropStatement, declarativeStrategy, "vtgate", "", "")
+		uuids = append(uuids, uuid)
+		onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusComplete)
+		onlineddl.CheckMigrationArtifacts(t, &vtParams, shards, uuid, true)
+		checkTable(t, tableName, false)
+	})
+	t.Run("revert DROP TABLE", func(t *testing.T) {
+		// This will recreate the table (well, actually, rename it back into place)
+		uuid := testRevertMigration(t, createRevertParams(uuids[len(uuids)-1]))
+		uuids = append(uuids, uuid)
+		onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusComplete)
+		checkTable(t, tableName, true)
+		checkMigratedTable(t, tableName, "create1")
+		testSelectTableMetrics(t)
+	})
+	t.Run("revert revert DROP TABLE", func(t *testing.T) {
+		// This will reapply DROP TABLE
+		uuid := testRevertMigration(t, createRevertParams(uuids[len(uuids)-1]))
+		uuids = append(uuids, uuid)
+		onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusComplete)
+		checkTable(t, tableName, false)
+	})
+	t.Run("declarative DROP TABLE where table does not exist", func(t *testing.T) {
+		uuid := testOnlineDDL(t, dropStatement, declarativeStrategy, "vtgate", "", "")
+		uuids = append(uuids, uuid)
+		onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusComplete)
+		onlineddl.CheckMigrationArtifacts(t, &vtParams, shards, uuid, false)
+		checkTable(t, tableName, false)
+	})
+	t.Run("revert DROP TABLE where table did not exist", func(t *testing.T) {
+		// Table will not be recreated because it didn't exist during the previous DROP TABLE
+		uuid := testRevertMigration(t, createRevertParams(uuids[len(uuids)-1]))
+		uuids = append(uuids, uuid)
+		onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusComplete)
+		checkTable(t, tableName, false)
+	})
+	// Table dropped. Let's start afresh.
+
+	// CREATE1
+	t.Run("declarative CREATE TABLE where table does not exist", func(t *testing.T) {
+		// The table does not exist
+		uuid := testOnlineDDL(t, createStatement1, declarativeStrategy, "vtgate", "create1", "")
+		uuids = append(uuids, uuid)
+		onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusComplete)
+		onlineddl.CheckMigrationArtifacts(t, &vtParams, shards, uuid, true)
+		checkTable(t, tableName, true)
+		initTable(t)
+		testSelectTableMetrics(t)
+	})
+	// CREATE2
+	t.Run("declarative CREATE TABLE with changes where table exists", func(t *testing.T) {
+		// The exists but with different schema
```

Comment:
Smallest of nits, but here and in the following sub-test comment we're missing `table` exists in the comment.

Reply:
fixed!